// @coreui/angular/esm2022/lib/header/header/header.component.mjs
import { Component, computed, input } from '@angular/core';
import { NgClass } from '@angular/common';
import * as i0 from "@angular/core";
export class HeaderComponent {
    constructor() {
        /**
         * Defines optional container wrapping children elements.
         */
        this.container = input();
        /**
         * Place header in non-static positions.
         */
        this.position = input();
        /**
         * Default role for header. [docs]
         * @type string
         * @default 'banner'
         */
        this.role = input('banner');
        this.hostClasses = computed(() => {
            return !!this.container() ? this.containerClasses() : this.headerClasses();
        });
        this.headerClasses = computed(() => {
            const position = this.position();
            return {
                header: true,
                [`header-${position}`]: !!position
            };
        });
        this.containerClasses = computed(() => {
            const container = this.container();
            return {
                container: container === true,
                [`container-${container}`]: typeof container === 'string'
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: HeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "18.2.12", type: HeaderComponent, isStandalone: true, selector: "c-header, [c-header]", inputs: { container: { classPropertyName: "container", publicName: "container", isSignal: true, isRequired: false, transformFunction: null }, position: { classPropertyName: "position", publicName: "position", isSignal: true, isRequired: false, transformFunction: null }, role: { classPropertyName: "role", publicName: "role", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "attr.role": "role()", "class": "hostClasses()" } }, ngImport: i0, template: "@if (!!container()) {\n  <div [ngClass]=\"headerClasses()\">\n    <ng-content />\n  </div>\n} @else {\n  <ng-content />\n}\n", dependencies: [{ kind: "directive", type: NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: HeaderComponent, decorators: [{
            type: Component,
            args: [{ selector: 'c-header, [c-header]', standalone: true, imports: [NgClass], host: { '[attr.role]': 'role()', '[class]': 'hostClasses()' }, template: "@if (!!container()) {\n  <div [ngClass]=\"headerClasses()\">\n    <ng-content />\n  </div>\n} @else {\n  <ng-content />\n}\n" }]
        }] });
