// @coreui/angular/esm2022/lib/callout/callout.component.mjs
import { Component, computed, input } from '@angular/core';
import * as i0 from "@angular/core";
export class CalloutComponent {
    constructor() {
        /**
         * Sets the color context of the component to one of CoreUI’s themed colors.
         * @type Colors
         */
        this.color = input();
        this.hostClasses = computed(() => {
            const color = this.color();
            return {
                callout: true,
                [`callout-${color}`]: !!color
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: CalloutComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.1.0", version: "18.2.12", type: CalloutComponent, isStandalone: true, selector: "c-callout, [cCallout]", inputs: { color: { classPropertyName: "color", publicName: "color", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()" }, classAttribute: "callout" }, ngImport: i0, template: '<ng-content />', isInline: true, styles: [":host{display:block}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: CalloutComponent, decorators: [{
            type: Component,
            args: [{ selector: 'c-callout, [cCallout]', template: '<ng-content />', standalone: true, host: { class: 'callout', '[class]': 'hostClasses()' }, styles: [":host{display:block}\n"] }]
        }] });
