// @coreui/angular/esm2022/lib/grid/container.component.mjs
import { booleanAttribute, Component, computed, input } from '@angular/core';
import * as i0 from "@angular/core";
export class ContainerComponent {
    constructor() {
        /**
         * Set container 100% wide until a breakpoint.
         */
        this.breakpoint = input('');
        /**
         * Set container 100% wide, spanning the entire width of the viewport.
         * @type InputSignalWithTransform<unknown, boolean>
         */
        this.fluid = input(false, {
            transform: booleanAttribute
        });
        this.hostClasses = computed(() => {
            const breakpoint = this.breakpoint();
            const fluid = this.fluid();
            return {
                container: !fluid && !breakpoint,
                'container-fluid': !!fluid,
                [`container-${breakpoint}`]: !!breakpoint
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: ContainerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.1.0", version: "18.2.12", type: ContainerComponent, isStandalone: true, selector: "c-container, [cContainer]", inputs: { breakpoint: { classPropertyName: "breakpoint", publicName: "breakpoint", isSignal: true, isRequired: false, transformFunction: null }, fluid: { classPropertyName: "fluid", publicName: "fluid", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()" } }, ngImport: i0, template: '<ng-content />', isInline: true, styles: [":host{display:block}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: ContainerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'c-container, [cContainer]', template: '<ng-content />', standalone: true, host: { '[class]': 'hostClasses()' }, styles: [":host{display:block}\n"] }]
        }] });
