// @coreui/angular/esm2022/lib/navbar/navbar-brand/navbar-brand.directive.mjs
import { Directive, input } from '@angular/core';
import * as i0 from "@angular/core";
export class NavbarBrandDirective {
    constructor() {
        this.role = input('button');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: NavbarBrandDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.1.0", version: "18.2.12", type: NavbarBrandDirective, isStandalone: true, selector: "[cNavbarBrand]", inputs: { role: { classPropertyName: "role", publicName: "role", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "attr.role": "role()" }, classAttribute: "navbar-brand" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: NavbarBrandDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cNavbarBrand]',
                    standalone: true,
                    host: { class: 'navbar-brand', '[attr.role]': 'role()' }
                }]
        }] });
