// @coreui/angular/esm2022/lib/accordion/accordion/accordion.component.mjs
import { booleanAttribute, Component, computed, effect, inject, input } from '@angular/core';
import { AccordionService } from '../accordion.service';
import * as i0 from "@angular/core";
export class AccordionComponent {
    constructor() {
        this.#accordionService = inject(AccordionService);
        /**
         * Removes the default background-color, some borders, and some rounded corners to render accordions edge-to-edge with their parent container.
         * @type boolean
         */
        this.flush = input(false, { transform: booleanAttribute });
        /**
         * Make accordion items stay open when another item is opened
         * @type boolean
         */
        this.alwaysOpen = input(false, { transform: booleanAttribute });
        this.#alwaysOpenEffect = effect(() => {
            this.#accordionService.alwaysOpen = this.alwaysOpen();
        });
        this.hostClasses = computed(() => ({
            accordion: true,
            'accordion-flush': this.flush()
        }));
    }
    #accordionService;
    #alwaysOpenEffect;
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: AccordionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.1.0", version: "18.2.12", type: AccordionComponent, isStandalone: true, selector: "c-accordion", inputs: { flush: { classPropertyName: "flush", publicName: "flush", isSignal: true, isRequired: false, transformFunction: null }, alwaysOpen: { classPropertyName: "alwaysOpen", publicName: "alwaysOpen", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()" } }, providers: [AccordionService], exportAs: ["cAccordionItem"], ngImport: i0, template: '<ng-content />', isInline: true, styles: [":host{display:block}\n"] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: AccordionComponent, decorators: [{
            type: Component,
            args: [{ selector: 'c-accordion', template: '<ng-content />', exportAs: 'cAccordionItem', providers: [AccordionService], standalone: true, host: { '[class]': 'hostClasses()' }, styles: [":host{display:block}\n"] }]
        }] });
