// @coreui/angular/esm2022/lib/form/input-group/input-group.component.mjs
import { Component, computed, input } from '@angular/core';
import * as i0 from "@angular/core";
export class InputGroupComponent {
    constructor() {
        /**
         * Size the component small or large.
         */
        this.sizing = input('');
        this.hostClasses = computed(() => {
            const sizing = this.sizing();
            return {
                'input-group': true,
                [`input-group-${sizing}`]: !!sizing
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: InputGroupComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.1.0", version: "18.2.12", type: InputGroupComponent, isStandalone: true, selector: "c-input-group", inputs: { sizing: { classPropertyName: "sizing", publicName: "sizing", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()" }, classAttribute: "input-group" }, ngImport: i0, template: '<ng-content />', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: InputGroupComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'c-input-group',
                    template: '<ng-content />',
                    standalone: true,
                    host: { class: 'input-group', '[class]': 'hostClasses()' }
                }]
        }] });
