// @coreui/angular/esm2022/lib/card/card-img.directive.mjs
import { computed, Directive, input } from '@angular/core';
import * as i0 from "@angular/core";
export class CardImgDirective {
    constructor() {
        /**
         * Optionally orientate the image to the top, bottom, or make it overlaid across the card.
         * @type {'top | 'bottom'}
         */
        this.orientation = input(undefined, { alias: 'cCardImg' });
        this.hostClasses = computed(() => {
            const orientation = this.orientation();
            const suffix = !!orientation ? `-${orientation}` : '';
            const horizontal = ['start', 'end'].includes(orientation ?? '-') ? orientation : undefined;
            return {
                [`card-img${suffix}`]: !horizontal,
                'img-fluid': !!horizontal,
                [`rounded-${horizontal}`]: !!horizontal
            };
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: CardImgDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.1.0", version: "18.2.12", type: CardImgDirective, isStandalone: true, selector: "[cCardImg]", inputs: { orientation: { classPropertyName: "orientation", publicName: "cCardImg", isSignal: true, isRequired: false, transformFunction: null } }, host: { properties: { "class": "hostClasses()" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.12", ngImport: i0, type: CardImgDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cCardImg]',
                    standalone: true,
                    host: { '[class]': 'hostClasses()' }
                }]
        }] });
